### PR TITLE
impl std::io::Write for SmallVec

### PR DESCRIPTION
**Pull request by chpio**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-smallvec/52)

### CODE REVIEW

**Comment by chpio** [comment]
this collides a little bit with the [*no_std* pr](https://github.com/servo/rust-smallvec/pull/49), because all other *std* modules are part of *core* but *io* is not.

**Comment by mbrubeck** [comment]
#49 has landed now, so this should be rebased on top of it, and `#[cfg(feature = "std")]` should be added to all the new code.

File: lib.rs [hunk 1]
```diff
@@ -1436,4 +1458,21 @@ pub mod tests {
         assert_eq!(&*small_vec, &[1, 2, 3, 4, 5]);
         drop(small_vec);
     }
+
+    #[cfg(feature = "std")]
```
Comment (chpio) [inline lib.rs]: Is that the correct way of doing this?

**Comment by mbrubeck** [comment]
@bors-servo r+

Yes, thanks!  When merged with #53, this builds and passes tests both with and without the `std` feature.

**Pull request merged**